src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -2930,9 +2930,10 @@ private void getCallInfo(ref CORINFO_RESOLVED_TOKEN pResolvedToken, CORINFO_RESO
             // Spec says that a callvirt lookup ignores static methods. Since static methods
             // can't have the exact same signature as instance methods, a lookup that found
             // a static method would have never found an instance method.
-            if (method.Signature.IsStatic && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_CALLVIRT) != 0)
+            if ((method.Signature.IsStatic && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_CALLVIRT) != 0)
+                || (method.IsNativeCallable && (flags & CORINFO_CALLINFO_FLAGS.CORINFO_CALLINFO_LDFTN) == 0))
```

Comment:
Could you break this up into two separate if blocks? The one for the static method with callvirt case should actually throw a different exception to be compatible with the CLR - it's just a little bit of extra work to construct the right exception and so we're just not completely compatible for the edge case https://github.com/dotnet/coreclr/blob/9da7cb7a2c9b5910260f2c838ac72114d00751ad/src/vm/jitinterface.cpp#L5189-L5195

Also, the comment above the `if` block would no longer apply for the new `if` block, which would be unfortunate.